GCRCatalogs/user_config_mgr.py

```diff
@@ -0,0 +1,127 @@
+import os
+import warnings
+import yaml
+from collections.abc import MutableMapping
+
+"""
+Utility class for managing user config (a dict) persisted to a yaml file. 
+"""
+
+all = ['UserConfigManager']
+
+
+class UserConfigManager(MutableMapping):
+    _default_config_filename = "gcr-catalogs.yaml"
+    _default_config_reldir = "lsstdesc"
+    
+    def __init__(self, config_filename=None, config_reldir=None):
+        config_name = os.path.basename(config_filename or self._default_config_filename)
+        reldir = config_reldir or self._default_config_reldir
+        if os.path.isabs(reldir):
+            raise ValueError(f'{reldir} must be a relative path')
+        self._user_config_dir = os.path.join(self._get_config_home(), reldir)
+        self._user_config_path = os.path.join(self._user_config_dir, config_name)
+
+    @staticmethod
+    def _get_config_home():
+        if os.getenv("XDG_CONFIG_HOME"):                   # Unix
+            return(os.getenv("XDG_CONFIG_HOME"))	
+        elif os.getenv("LOCALAPPDATA"):                     # Win
+            return(os.getenv("LOCALAPPDATA"))
+
+        return os.path.join(os.path.expanduser("~"), ".config")
+
+    # Override methods which may access more than one item for efficiency
+    def keys(self):
+        return self._load_config().keys()
+
+    def values(self):
+        return self._load_config().values()
+
+    def items(self):
+        return self._load_config().items()
+
+    def update(self, items):
+        """
+        Write one or more key-value pairs to the config file.  Create new file or
+        append to existing file. 
+
+        Parameters
+        ----------
+        items       dict of entries to be written to file
+
+        Returns
+        -------
+        dict written to the file.  
+        If overwrite condition is violated, raise ValueError
+
+        """
+        config_dict = {}
+
+        if self._user_config_exists():
+            config_dict = self._load_config()
+
+        config_dict.update(items)
+        return self._write_config(config_dict)
+    
+    def clear(self):
+        return self._write_config(dict())
+
+    def deleteitems(self, keys, absent_ok=True):
+        """
+        Remove specified items from config file
+
+        Parameters
+        ----------
+        A collection of keys
+        """
+
+        config_dict = self._load_config()
+
+        keys_to_try = keys
+        if absent_ok:
+            keys_to_try = set(keys).intersection(config_dict.keys())
+
+        for k in keys_to_try:
+            del config_dict[k]
```

Comment:
Right now when `absent_ok` is set to false, some keys may be deleted (the ones before the first missing key), while other will not. I am not sure if this is what a user would expected, or if it has a common use case. I would suggest that we do no action if `absent_ok` is set to false and not all keys are in `config_dict`.

```suggestion
        if not absent_ok and not all(key in config_dict for key in keys):
        	raise KeyError("Some keys do not exist")

        for key in keys:
            config_dict.pop(key, None)
```

Reply:
Thanks for catching this.  Indeed that behavior is not what anyone would expect.